app/assets/javascripts/quby/answers/validation.js

```diff
@@ -189,6 +208,57 @@
     return !failed;
   };
 
+  function parsePartialDate(inputs) {
+    var values = dateValuesWithDefaults(inputs)
+    if(!dateValuesValid(values)) {
+      throw "invalidDate";
+    }
+    
+    return new Date(values.year, values.month, values.day, values.hour, values.minute);
+  }
+
+  function dateValuesValid(values) {
+    return values['year']   >= 1900 && values['year']   <= 2100 &&
+           values['month']  >= 0    && values['month']  <= 11   &&
+           values['day']    >= 1    && values['day']    <= 31   &&
+           values['hour']   >= 0    && values['hour']   <= 23   &&
+           values['minute'] >= 0    && values['minute'] <= 59;
+  }
+
+  function dateValuesWithDefaults(inputs) {
+    var value = function(placeholder) {
+      var val = inputs.filter("[placeholder=" + placeholder + "]").first().val();
+      if(val === undefined || val == "")
+        return null;
+      if(intVal = parseInt(val)) {
+        return intVal;
+      } else {
+        throw "invalidDate";
+      }
+    };
+
+    return {
+      year:   value('YYYY') || 2000,
+      month:  value('MM')   ? value('MM')-1 : 0,    // JS months range from 0-11 instead of 1-12
+      day:    value('DD')   || 1,
```

Comment:
Eigenlijk converteer je hier 0 naar 1, aangezien 0 falsy is in javascript. In deze methode zorgt dat niet voor vervelende effecten, maar onthoud dat `||` gebruiken om aanwezigheid te testen in javascript deze valkuil heeft.